subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy

```diff
@@ -362,4 +365,49 @@ public class StaticInnerTest {
         'includeEngines' | '"junit-jupiter"'
         'excludeEngines' | '"junit-jupiter"'
     }
+
+    @Issue('https://github.com/gradle/gradle/issues/6453')
+    def "can handle parallel test execution"() {
+        given:
+        def numTestClasses = 2
```

Comment:
Do 2 tests consistently reproduce the previous issue? I think a few dozen wouldn't make this test much longer and might trigger concurrency issues more reliably.

Reply:
They did. But you're right, adding more doesn't hurt, so I will.